src/utils/registerDockerContextStatusBarItems.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IActionContext, registerEvent } from '@microsoft/vscode-azext-utils';
+import * as vscode from 'vscode';
+import { ext } from "../extensionVariables";
+import { ListContextItem } from '../runtimes/docker/contracts/ContainerClient';
+
+const dockerContextStatusBarSetting = 'contexts.showInStatusBar';
+
+export function registerDockerContextStatusBarEvent(ctx: vscode.ExtensionContext): void {
+    // Register an event to watch for changes to config, reconfigure if needed
+    registerEvent('docker.context.showInStatusBar.changed', vscode.workspace.onDidChangeConfiguration, (actionContext: IActionContext, e: vscode.ConfigurationChangeEvent) => {
+
+        actionContext.telemetry.suppressAll = true;
+        actionContext.errorHandling.suppressDisplay = true;
+
+        if (e.affectsConfiguration('docker.contexts.showInStatusBar')) {
+            // Don't wait
+            void scheduleUpdateStatusBar();
+        }
+    });
+
+    ctx.subscriptions.push(
+        ext.dockerContextStatusBarItem,
+        ext.runtimeManager.contextManager.onContextChanged(scheduleUpdateStatusBar)
+    );
+
+    // Don't wait
+    void scheduleUpdateStatusBar();
+}
+
+export async function showStatusBarItemIfNeeded() {
```

Comment:
```suggestion
async function showStatusBarItemIfNeeded() {
```
Since this function is not being used externally we don't need to export it